packages/promotion/src/models/campaign.ts

```diff
@@ -0,0 +1,101 @@
+import { generateEntityId } from "@medusajs/utils"
+import {
+  BeforeCreate,
+  Collection,
+  Entity,
+  OnInit,
+  OneToMany,
+  OneToOne,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+  Unique,
+} from "@mikro-orm/core"
+import CampaignBudget from "./campaign-budget"
+import Promotion from "./promotion"
+
+type OptionalFields =
+  | "description"
+  | "currency"
+  | "created_at"
+  | "updated_at"
+  | "deleted_at"
+
+type OptionalRelations = "budget"
+
+@Entity()
+export default class Campaign {
+  [OptionalProps]?: OptionalFields | OptionalRelations
+
+  @PrimaryKey({ columnType: "text" })
+  id!: string
+
+  @Property({ columnType: "text" })
+  name: string
+
+  @Property({ columnType: "text", nullable: true })
+  description?: string
```

Comment:
Couple of nullable properties should be `| null` here. 

if you want them to be optional that's fine, then `starts_at` and `ends_at` should also be optional for consistency
```suggestion
  description: string | null
```